modifytags/modifytags.go

```diff
@@ -0,0 +1,376 @@
+	PascalCase            // MyField -> MyField
+	TitleCase             // MyField -> My Field
+	Keep                  // keep the existing field name
+)
+
+// A Modification defines how struct tags should be modified for a given input struct.
+type Modification struct {
+	Add        []string            // tags to add
+	AddOptions map[string][]string // options to add, per tag
+
+	Remove        []string            // tags to remove
+	RemoveOptions map[string][]string // options to remove, per tag
+
+	Overwrite            bool // if set, replace existing tags when adding
+	SkipUnexportedFields bool // if set, do not modify tags on unexported struct fields
+
+	Transform    Transform // transform rule for adding tags
+	Sort         bool      // if set, sort tags in ascending order by key name
+	ValueFormat  string    // format for the tag's value, after transformation; for example "column:{field}"
+	Clear        bool      // if set, clear all tags. tags are cleared before any new tags are added
+	ClearOptions bool      // if set, clear all tag options; options are cleared before any new options are added
+}
+
+// Apply applies the struct tag modifications of the receiver to all
+// struct fields contained within the given node between start and end position, modifying its input.
+func (mod *Modification) Apply(fset *token.FileSet, node ast.Node, start, end token.Pos) error {
+	err := mod.validate()
+	if err != nil {
+		return err
+	}
+
+	err = mod.rewrite(fset, node, start, end)
+	if _, ok := err.(*RewriteErrors); ok {
+		return err
+	}
+	return nil
+}
+
+// rewrite rewrites the node for structs between the start and end positions
+func (mod *Modification) rewrite(fset *token.FileSet, node ast.Node, start, end token.Pos) error {
+	var errs []error
+
+	rewriteFunc := func(n ast.Node) bool {
+		x, ok := n.(*ast.StructType)
+		if !ok {
+			return true
+		}
+
+		for _, f := range x.Fields.List {
+			pos := f.Pos()
+			if !(start <= pos && pos <= end) {
+				continue // not in range
+			}
+
+			fieldName := ""
+			if len(f.Names) != 0 {
+				for _, field := range f.Names {
+					if !mod.SkipUnexportedFields || isPublicName(field.Name) {
+						fieldName = field.Name
+						break
+					}
+				}
+			}
+
+			// anonymous field
+			if f.Names == nil {
+				ident, ok := f.Type.(*ast.Ident)
+				if !ok {
+					continue
+				}
+
+				if !mod.SkipUnexportedFields {
+					fieldName = ident.Name
+				}
+			}
+
+			// nothing to process, continue with next field
+			if fieldName == "" {
+				continue
+			}
+
+			if f.Tag == nil {
+				f.Tag = &ast.BasicLit{}
+			}
+
+			res, err := mod.processField(fieldName, f.Tag.Value)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("%s:%d:%d:%s",
+					fset.Position(f.Pos()).Filename,
+					fset.Position(f.Pos()).Line,
+					fset.Position(f.Pos()).Column,
+					err))
+				continue
+			}
+
+			f.Tag.Value = res
+		}
+
+		return true
+	}
+
+	ast.Inspect(node, rewriteFunc)
+
+	rewriteErrs := &RewriteErrors{Errs: errs}
```

Comment:
Invert this, both to follow the best practice of [indenting error flow](https://google.github.io/styleguide/go/decisions#indent-error-flow), and to avoid the rewriteErrs variable.

```go
if len(errs) > 0 {
  return &RewriteErrors{Errs: errs}
}
return nil
```